imap_processing/cdf/tests/test_cdf_attribute_manager.py

```diff
@@ -203,19 +200,31 @@ def test_add_global_attribute():
 
     # Changing a dynamic global variable
     cdf_manager.add_global_attribute("Project", "Test Project")
-    assert cdf_manager.global_attributes["Project"] == "Test Project"
     test_get_global_attrs = cdf_manager.get_global_attributes("imap_test_T1_test")
+    assert cdf_manager.global_attributes["Project"] == "Test Project"
     assert test_get_global_attrs["Project"] == "Test Project"
 
     # Testing adding required global attribute
     cdf_manager.global_attributes.__delitem__("Source_name")
+    # Reloading get_global_attributes to pick up deleted Source_name
+    test_get_global_attrs = cdf_manager.get_global_attributes("imap_test_T1_test")
     with pytest.raises(KeyError):
-        assert cdf_manager.global_attributes["Source_name"] == "Wrong"
-    assert cdf_manager.global_attributes["Source_name"] is None
-
+        assert cdf_manager.global_attributes["Source_name"] == "Does_not_exist"
+    assert test_get_global_attrs["Source_name"] is None
+    # Adding deleted global attribute
     cdf_manager.add_global_attribute("Source_name", "anas_source")
     assert cdf_manager.global_attributes["Source_name"] == "anas_source"
```

Comment:
Nice, I'd also add:
```suggestion
    assert cdf_manager.global_attributes["Source_name"] == "anas_source"
    test_get_global_attrs = cdf_manager.get_global_attributes("imap_test_T1_test")
    assert test_get_global_attrs["Source_name"] == "anas_source"
```

but that is definitely a nitpick!